search/search_quality/assessment_tool/main_model.cpp

```diff
@@ -383,6 +393,21 @@ void MainModel::ShowMarks(Context const & context)
                                    context.m_nonFoundResultsEdits.GetEntries());
 }
 
+void MainModel::OnChangeAllRelevancesClicked(Edits::Relevance relevance)
+{
+  CHECK_GREATER_OR_EQUAL(m_selectedSample, 0, ());
+  CHECK_LESS(m_selectedSample, m_contexts.Size(), ());
+  auto & context = m_contexts[m_selectedSample];
+
+  context.m_foundResultsEdits.SetAllRelevances(relevance);
+  context.m_nonFoundResultsEdits.SetAllRelevances(relevance);
+
+  m_view->OnResultChanged(m_selectedSample, View::ResultType::Found, Edits::Update::MakeAll());
+  m_view->OnResultChanged(m_selectedSample, View::ResultType::NonFound, Edits::Update::MakeAll());
```

Comment:
What the reason to call OnResultChanged for both Found and NonFound?

Reply:
My reasoning was "it won't hurt and will be harder to forget when refactoring".
Currently,
1. A click on the button modifies relevances both for found and non-found results, so let's inform both.
2. These two lines are essentially nop, the next two are enough.